ProjectMars/StepDefinitions/LanguageFeatureStepDefinitions.cs

```diff
@@ -0,0 +1,175 @@
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using ProjectMars.Pages;
+using ProjectMars.Utilities;
+using System;
+using System.Security.Policy;
+using TechTalk.SpecFlow;
+
+namespace ProjectMars.StepDefinitions
+{
+    [Binding]
+    
+    public class LanguageFeatureStepDefinitions:CommonDriver
+    {
+        LoginPage loginPageObj;
+        ProfilePage profilePageObj;
+
+        [Given(@"I logged into Mars portal sucessfully")]
+        public void GivenILoggedIntoMarsPortalSucessfully()
+        {
+            loginPageObj = new LoginPage(driver);
+            loginPageObj.LoginActions(driver, "antususansunila@gmail.com", "Ammuj@me5");
+        }
+
+        [Given(@"I navigate to Profile page")]
+        public void GivenINavigateToProfilePage()
+        {
+            profilePageObj = new ProfilePage(driver);
+            profilePageObj.GotoLanguageTab(driver);
+        }
+
+        [When(@"I add new '([^']*)' and '([^']*)'")]
+        public void WhenIAddNewAnd(string p0, string p1)
+        {
+            profilePageObj.ResetLanguageRow(driver);
+            profilePageObj.CreateLanguageRecord(driver,p0,p1);
+            
+        }
+
+        [Then(@"the record should have new '([^']*)' and '([^']*)'")]
+        public void ThenTheRecordShouldHaveNewAnd(string p0, string p1)
+        {
+            string newLanguage = profilePageObj.GetLanguage(driver);
+            string newLevel = profilePageObj.GetLevel(driver);
+            Assert.That(newLanguage == p0, "Language has not been created sucessfully.");
+            Assert.That(newLevel == p1, "Level has not created sucessfully");
+        }
+
+
+        [When(@"I add invalid '([^']*)' and valid '([^']*)'")]
+        public void WhenIAddInvalidAndValid(string p0, string p1)
+        {
+            profilePageObj.ResetLanguageRow(driver);
+            profilePageObj.CreateLanguageRecord(driver, p0, p1);
+        }
+
+        [Then(@"the record should  not be added '([^']*)' sucessfully")]
+        public void ThenTheRecordShouldNotBeAddedSucessfully(string p2)
+        {
+            
+            string actualResult1 = driver.FindElement(By.XPath("//*[@class='ns-box-inner']")).Text;
```

Comment:
please move this within the page, create a method that returns the value and then use it! best practice to contain page logic within pages